<?xml version="1.0"?>
<launch>

<node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
  <param name="output_frame" value="odom_combined"/>
  <param name="base_footprint_frame" value="base_link"/>
  <param name="freq" value="100.0"/>
  <param name="sensor_timeout" value="0.1"/>  
  <param name="odom_used" value="true"/>
  <param name="imu_used" value="true"/>
  <param name="vo_used" value="false"/>

  <remap from="odom" to="odometry_covar" />
  <remap from="imu_data" to="imu/data" />
</node>

<include file="$(find xsens_driver)/launch/xsens_driver.launch">
<arg name="device" value="/dev/ttyUSB1"/>
</include>

<node pkg="odometry_correction" type="odometry_add_covar.py" name="odometry_add_covar">
<remap from="odometry_input" to="odometrija"/>
<remap from="odometry_output" to="odometry_covar"/>
</node>

<node pkg="odometry_correction" type="pose_to_odometry_topic.py" name="pose_to_odometry_topic"/>
<remap from="odometry_input" to="/robot_pose_ekf/odom_combined"/>
<remap from="odometry_output" to="odometry_fused"/>
</node>

<node pkg="tf" type="static_transform_publisher" name="imu_broadcaster" args="0.0 0 0.0 0 0 0 base_link imu 100" /> 


</launch>

